cogeo_mosaic/backends/dynamodb.py

```diff
@@ -0,0 +1,140 @@
+import functools
+import itertools
+import json
+import logging
+import os
+from decimal import Decimal
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import boto3
+import mercantile
+
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import find_quadkeys
+from cogeo_mosaic.model import MosaicJSON
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class DynamoDBBackend(BaseBackend):
+    """DynamoDB Backend Adapter"""
+
+    def __init__(
+        self,
+        table_name: str,
+        mosaic_def: Optional[Union[MosaicJSON, Dict]] = None,
+        region: str = os.getenv("AWS_REGION", "us-east-1"),
+        client: Optional[Any] = None,
+    ):
+        self.client = client or boto3.resource("dynamodb", region_name=region)
+        self.table = self.client.Table(table_name)
+
+        if mosaic_def is not None:
+            self.mosaic_def = MosaicJSON(**dict(mosaic_def))
+        else:
+            self.mosaic_def = self.read_mosaic()
+
+    def tile(self, x: int, y: int, z: int) -> Tuple[str]:
+        """Retrieve assets for tile."""
+        return self.get_assets(x, y, z)
+
+    def point(self, lng: float, lat: float) -> Tuple[str]:
+        """Retrieve assets for point."""
+        tile = mercantile.tile(lng, lat, self.quadkey_zoom)
+        return self.get_assets(tile.x, tile.y, tile.z)
+
+    def upload(self):
+        self._create_table()
+        items = self._create_items()
+        self._upload_items(items)
+
+    def update(self):
+        raise NotImplementedError
+
+    def _create_table(self, billing_mode: str = "PAY_PER_REQUEST"):
```

Comment:
maybe we could use `enum` for billing_mode, so we make sure we only pass good values

Reply:
I have to look up how to do this, and this option currently isn't exposed to the user at all